rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,501 @@
+- (+) Takes up less vertical space
+- (-) Visually confusing when every line starts with a reserved character.
+    - (-) Potential visual conflicts with the content
+- (+) Indentation of the content is clear.
+
+</details>
+
+## Doc-comment examples
+
+The following examples demonstrate the concrete usage scenarios:
+
+`somefile.nix`
+
+~~~nix
+{
+  /**
+  Documentation for mapAttrs
+  
+  # Example
+  
+  ```
+  mapAttrs {attr = "foo"; } ...
+  ```
+  
+  # Type
+  
+  ```
+  mapAttrs :: a -> b -> c
+  ```
+  
+  */
+  mapAttrs = f: s: <...>;
+}
+~~~
+
+## Examples
+
+This section contains examples for the different use cases we would end up with; Visualize them and emphasize the previously discussed characteristics.
+
+### Attribute bindings 
+
+`somefile.nix`
+
+~~~nix
+{
+  /**
+  mapAttrs is a well-known function 
+  
+  # Examples
+  
+  "`Nix
+  some code examples
+  ```
+  
+  */
+  mapAttrs = f: s: #...
+}
+~~~
+
+### NixOS Module documentation 
+
+> Note: NixOS Modules also can produce documentation from their interface declarations. However, this does not include a generic description and usage examples.
+
+`myModule.nix`
+~~~nix
+/**
+  This is a custom module.
+  
+  It configures a systemd service.
+  
+  # Examples
+  
+  Different use case scenarios
+  how to use this module
+  
+*/
+{config, ...}:
+{
+  config = f: s: f s;
+}
+~~~
+
+### Anonymous function documentation 
+
+`function.nix`
+~~~nix
+/**
+  This is an anonymous function implementation.
+  
+  It does not have a name yet. 
+  Nevertheless, documentation can be right next to the implementation.
+  
+  The name gets assigned later:
+  
+  "`Nix
+  {
+     plus = import ./function.nix;
+  }
+  ```
+  
+  (Future) native Nix or community tools provide 
+  implementations to track doc-comments within the Nix evaluator.
+  Documentation of `sum` can then be inferred. 
+  This still needs to be specified/implemented!
+
+*/
+{a, b}:
+{
+  sum = a + b;
+}
+~~~
+
+### Anonymous expression documentation 
+
+`exp.nix`
+~~~nix
+/**
+  This is an anonymous string.
+ 
+  It is the documentation for the "anonymous expression" use case.
+  
+  Although this example is relatively superficial, there might be use cases.
+*/
+"-p=libxml2/include/libxml2"
+~~~
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+
+## Changes the existing comments inside the code base.
+
+This could be automated. (e.g., via codemod)
+
+Also, this affects only the `lib` folder and a few other places currently used to build the documentation.
+
+## Changes in the `nixdoc' tooling are required. 
+
+It remains an open question that needs to be tried;
+
+Can we still build the current Nixos manuals with the new standard?
+
+- We can wait to change everything.
+- Migration can happen in small steps.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## All considered outer formats
+    
+| Property / Approach | `##` | `/** */` | `Javadoc` | `/*\|` or `/*^`  |
```

Comment:
It might be some of this here actually breaks rendering of the rest of the text?

Reply:
Where do you have rendering issues? I am viewing it in GitHub and in vs-code. Both look good. 

![image](https://user-images.githubusercontent.com/50398876/235355957-fa67964b-ed7c-4577-8dbf-3ddb1b67b934.png)

Reply:
This was in the "Files changed" view, when looking at the literal markdown and its syntax highlighting. It might be just a rendering bug on GitHub's side, if it renders fine.

Reply:
Ah, syntax highlighting is bugged sometimes. It doesn't look at the real markdown AST, it is just regex under the hood. Which fails to highlight this correctly I guess.

Reply:
I think syntax highlighting was affected by unbalanced ticks/quotes. See [this comment](https://github.com/NixOS/rfcs/pull/145/files/c30cb3331c15c4e6980150c4a5a07e1fc246eeac#r1181132755). Right now it looks ok in diff.